app/workers/asset_manager_create_whitehall_asset_worker.rb

```diff
@@ -23,8 +23,17 @@ def perform(file_path, legacy_url_path, draft = false, model_class = nil, model_
     asset_manager.create_whitehall_asset(asset_options)
 
     if model
+      # sadly we can't just search for url, because it's a magic
+      # carrierwave thing not in our model
       Attachment.where(attachable: model.attachables).where.not(attachment_data: nil).find_each do |attachment|
-        attachment.attachment_data.uploaded_to_asset_manager!
+        # 'attachment.attachment_data' can still be nil even with the
+        # check above, because if the 'attachment_data_id' is non-nil
+        # but invalid, the 'attachment_data' will be nil - and the
+        # generated SQL only checks if the 'attachment_data_id' is
+        # nil.
+        if attachment.attachment_data && attachment.attachment_data.url == legacy_url_path
```

Comment:
Is `url` a field in the database? Can we do something like `.where(attachment_data: { url: legacy_url_path })` in the query?

Reply:
Sad about that having to include that `attachment_data` check :( Wonder if there are other places where we have a similar problem.

Reply:
Sadly `AttachmentData.url` is not a thing.  The closest is `carrierwave_file`, which is the basename of the file.  So an alternative would be looking up all `AttachmentData`s with the right basename and then checking their `url`.

Reply:
Although `carrierwave_file` doesn't seem to be find thumbnails.